guides/source/debugging_rails_applications.md

```diff
@@ -849,6 +800,63 @@ set forcestep
 set listsize 25
 ```
 
+Debugging with the `web-console` gem
+------------------------------------
+
+Web Console is a bit like `byebug`, but it runs in the browser. In any page you
+are developing, you can request a console in the context of a view or a
+controller. The console would be rendered next to your HTML content.
+
+### Console
+
+Inside any controller action or view, you can then invoke the console by
+calling the `console` method.
+
+For example, in a controller:
+
+```ruby
+class PostController < ApplicationController
+  def new
+    console
+    @post = Post.new
+  end
+end
+```
+
+Or in a view:
+
+```html+erb
+<% console %>
+
+<h2>New Post</h2>
+```
+
+This will render a console in the browser on top of the post creation HTML
+view. You don't need to care about the `console` output, the console won't be
+rendered on the spot of the `console` invocation.
+
+The console executes pure Ruby code. You can define and instantiate
+custom classes, create new models and inspect variables.
+
+NOTE: Only one console can be rendered per request. Otherwise `web-console`
+will raise an error on the second `console` invocation.
+
+### Inspecting Variables
+
+You can invoke `instance_variables` to list all the instance variables
+available in your context. If you want to list all the local variables, you can
+do that with `local_variables`.
+
+### Settings
+
+* `config.web_console.whitelisted_ips`: Authorized list of IPv4 or IPv6
+addresses and networks (default: `127.0.0.1/8, ::1`).
```

Comment:
I followed the `byebug` format above. I'll quote, since I can't easily link to the markdown line on GH: `set autoreload: Reload source code when changed (default: true).`. If we are to change this format, I think we should change the `byebug` one as well.

WDYT @robin850, @vipulnsward?